pyqir-jit/src/jit.rs

```diff
@@ -27,22 +31,30 @@ pub fn run_module_file(
         .to_str()
         .expect("Did not find a valid Unicode path string")
         .to_owned();
-    let context_type = ContextType::File(&path_str);
-    let context = BareContext::new(&ctx, context_type)?;
+
+    let module_type = ModuleType::File(&path_str);
+    let context = Bare::new(&ctx, module_type)?;
     run_module(&context.module, entry_point)
 }
 
-pub fn run_module<'ctx>(
-    module: &Module<'ctx>,
-    entry_point: Option<&str>,
-) -> Result<SemanticModel, String> {
-    Target::initialize_native(&InitializationConfig::default()).unwrap();
+/// # Errors
+///
+/// Will return `Err` if LLVM native target fails to initialize
+pub fn run_module(module: &Module<'_>, entry_point: Option<&str>) -> Result<SemanticModel, String> {
+    Target::initialize_native(&InitializationConfig::default())
+        .expect("Failed to initialize native target.");
```

Comment:
The comment implies that this shouldn't panic but instead return the error.

Reply:
The Errors does concern the associated message. It also can panic, but clippy didn't catch the change. Should we add a return error instead and update the doc?

Reply:
I'm not sure I understand - if you change this line to a return an error, then I don't think the doc needs to be updated, because this error case falls under "return `Err` if LLVM native target fails to initialize". However, if you leave it as a panic, then I think the doc needs to be updated.